apps/dsb-message-broker/src/channel/dto/create-channel.dto.ts

```diff
@@ -1,53 +1,91 @@
     })
     @IsString()
     @IsNotEmpty()
     fqcn: string;
 
     @ApiProperty({
-        type: Object,
-        description: 'Array of topics that is available in the channel'
+        type: 'object',
+        properties: {
+            namespace: {
+                type: 'string'
+            },
+            schema: {
+                type: 'string | JSONSchemaType'
+            }
+        },
+        required: false,
+        description: 'Array of topic objects that determines topics for messages.',
+        example: {
+            namespace: 'topic1',
+            schema: '{"type":"object","properties":{"foo":{"type":"integer"},"bar":{"type":"string"}},"required":["foo"],"additionalProperties":false}'
+        }
     })
     @IsOptional()
     topics?: {
         namespace: string;
-        schemaType: 'json' | 'xml' | 'JSON' | 'XML';
-        schema: string;
+        schema: JSONSchemaType<any> | string;
     }[];
 
     @ApiProperty({
-        type: Array,
+        isArray: true,
+        required: false,
         description:
-            'Array of DIDs and roles that have permision to publish messages to the channel. If it is ommited, any user with "user" role can publish messages to the channel.'
+            'Array of DIDs that have permision to edit the channel. If it is ommited, creator of the channel will be the default admin.',
+        example: ['did:ethr:0x5aEa5Bf5c5b341A0BFhryv5b51b77Fb9807F1b52']
+    })
+    @IsOptional()
+    admins?: string[];
+
+    @ApiProperty({
+        isArray: true,
+        required: false,
+        description:
+            'A mixed array of DIDs and roles that have permision to publish messages to the channel. If it is ommited, any user with "user" role can publish messages to the channel.',
+        example: [
+            'did:ethr:0x5aEa5Bf5c5b341A0BFhryv5b51b77Fb9807F1b52',
+            'user.roles.dsb.apps.energyweb.iam.ewc'
+        ]
     })
     @IsOptional()
     publishers?: string[];
 
     @ApiProperty({
-        type: Array,
+        isArray: true,
+        required: false,
         description:
-            'Array of DIDs and roles that have permision to subscribe to the channel. If it is ommited, any user with "user" role can subscribe to the channel.'
+            'A mixed array of DIDs and roles that have permision to subscribe to the channel. If it is ommited, any user with "user" role can subscribe to the channel.',
+        example: [
+            'did:ethr:0x5aEa5Bf5c5b341A0BFhryv5b51b77Fb9807F1b52',
+            'user.roles.dsb.apps.energyweb.iam.ewc'
+        ]
     })
     @IsOptional()
     subscribers?: string[];
 
     @ApiProperty({
-        type: Number,
-        description: 'Maximum age of any message in the channel, expressed in nanoseconds.'
+        type: 'number',
+        required: false,
+        description: 'Maximum age of any message in the channel, expressed in nanoseconds.',
+        example: 24 * 60 * 60 * 1000 * 1000
```

Comment:
I meant the calculated number (86400000000 - e.g. 24 hours)

Reply:
edited